fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_SecFplnMain.js

```diff
@@ -14,41 +14,92 @@ class CDUSecFplnMain {
             mcdu.efisInterfaces.R.setSecRelatedPageOpen(false);
         };
 
+        const hasSecondary = mcdu.flightPlanService.hasSecondary(1);
+
+        const deleteSecColumn = new Column(0, "");
+
+        // <DELETE SEC
+        if (hasSecondary) {
+            deleteSecColumn.update("{DELETE SEC");
+
+            mcdu.onLeftInput[2] = () => {
+                mcdu.flightPlanService.secondaryReset(1);
+            };
+        }
+
+        const activateSecColumn = new Column(0, "");
+
+        const activePlan = mcdu.flightPlanService.active;
+
+        let canCopyOrSwapSec = false;
+        if (hasSecondary) {
+            const secPlan = mcdu.flightPlanService.secondary(1);
+
+            const activeToLeg = activePlan.activeLeg;
+            const secToLeg = secPlan.activeLeg;
+
+            canCopyOrSwapSec = !mcdu.navModeEngaged() || (Fmgc.FlightPlanUtils.areFlightPlanElementsSame(activeToLeg, secToLeg) && activePlan.activeLegIndex === secPlan.activeLegIndex);
+        }
+
+        // *ACTIVATE SEC
+        if (hasSecondary && canCopyOrSwapSec) {
```

Comment:
I think `hasSecondary` is redundant as `canCopyOrSwapSec` can only be true if `hasSecondary` is true.

Reply:
Done.